README.md

```diff
@@ -1349,6 +1349,31 @@ _You can enable the following settings in Xcode by running [this script](resourc
 
 * <a id='newline-at-eof'></a>(<a href='#newline-at-eof'>link</a>) **Files should end in a newline.** [![SwiftLint: trailing_newline](https://img.shields.io/badge/SwiftLint-trailing__newline-007A87.svg)](https://github.com/realm/SwiftLint/blob/master/Rules.md#trailing-newline)
 
+* <a id='mark-types-and-extensions'></a>(<a href='#mark-types-and-extensions'>link</a>) **Each type and extension should be preceded by a `MARK` comment.** [![SwiftFormat: markTypes](https://img.shields.io/badge/SwiftFormat-markTypes-008489.svg)](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#markTypes)
+  * Types should be preceded by a `// MARK: - TypeName` comment.
+  * Extensions that add a conformance should be preceded by a `// MARK: TypeName + ProtocolName` comment.
+  * Extensions that follow the type being extended should omit that type's name and instead use `// MARK: ProtocolName`.
+  * If there is only one type or extension in a file, the `MARK` comment can be omitted.
+  * For extensions that do not add new conformances, consider adding a `// MARK:` with a descriptive comment.
```

Comment:
It's not really possible to autocorrect a descriptive `MARK` comment for extensions that don't add additional conformances, but it seems valuable to leave this in as a suggestion / best-practice.

Reply:
How does the autocorrect rule handle this?

Reply:
This can't be autocorrected, since the linter has no way of synthesizing a useful "descriptive comment". It seems useful to include optional guidance (_"consider adding"_) for what to do in this case.

An example of this in practice would be something like:

```swift
// MARK: Constraint helpers

extension UIView {
  func constrainToSuperview() { ... }
  func constrainToSuperviewSafeAreaInsets() { ... }
}
```

Reply:
What does SwiftFormat do if I wrote something like this?

```swift

// MARK: Constraint helpers

extension UIView: RandomProtocol {
  func constrainToSuperview() { ... }
  func constrainToSuperviewSafeAreaInsets() { ... }
}
```

I'm curious how autocorrection works in that case.

Reply:
The SwiftFormat rule will replace any existing mark with the correct mark. In this case: 

```diff
- // MARK: Constraint helpers
+ // MARK: UIView + RandomProtocol

 extension UIView: RandomProtocol {
   func constrainToSuperview() { ... }
   func constrainToSuperviewSafeAreaInsets() { ... }
 }
```